Lodash import issue in v4

### Astro Info

```block
Astro                    v4.0.3
Node                     v18.16.1
System                   Windows (x64)
Package Manager          npm
Output                   static
Adapter                  none
Integrations             @astrojs/react
                         @astrojs/tailwind
```


### If this issue only occurs in one browser, which browser is a problem?

All browsers

### Describe the Bug

Updated astro to v4, run the project, got an immediate error : 

    TypeError: __vite_ssr_import_0__.isObject is not a function

isObject being imported through
 
    import { isObject } from "lodash";

I also tried to remove node_modules and package-lock and reinstalling, same result. Also tried a new project, same result. Then tried on stackblitz, same result.

I tried to add lodash to my astro.config.mjs as such : 

    export default defineConfig({
      integrations: [react(), tailwind()],
      vite: {
        ssr: {
          external: ["lodash"]
        }
      }
    });

No idea if that's supposed to work as I never dwelved into vite before.

Of course lodash is properly referenced in my package.json file and present in node_modules. 

    {
      "dependencies": {
        "lodash": "^4.17.21",
       }
    }

Also : 

    PS G:\fo> npm list lodash
    fo@0.0.1 G:\fo
    └── lodash@4.17.21

### What's the expected result?

I expect it to work properly.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-duffch?file=src%2Fcomponents%2FCard.astro

### Participation

- [ ] I am willing to submit a pull request for this issue.

Seems like a Vite bug where a proper error isn't emitted. I'll send a fix soon, but ideally it should have shown this error instead:

```
16:35:47 [ERROR] [vite] Named export 'isObject' not found. The requested module 'lodash' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from 'lodash';
const {isObject} = pkg;

  Stack trace:
    at /Users/bjorn/Work/repros/astro-lodash/src/components/Card.astro:4:31
    [...] See full stack trace in the browser, or rerun with --verbose.
```

`lodash-es` would be better in this case either ways as it can be treeshaken.

--- a/packages/vite/src/node/ssr/ssrModuleLoader.ts
+++ b/packages/vite/src/node/ssr/ssrModuleLoader.ts
@@ -291,8 +291,8 @@ async function nodeImport(
   metadata?: SSRImportMetadata,
 ) {
   let url: string
-  const isRuntimeHandled = id.startsWith('data:') || isBuiltin(id)
-  if (isRuntimeHandled) {
+  let filePath: string | undefined
+  if (id.startsWith('data:') || isBuiltin(id)) {
     url = id
   } else {
     const resolved = tryNodeResolve(
@@ -310,6 +310,7 @@ async function nodeImport(
       err.code = 'ERR_MODULE_NOT_FOUND'
       throw err
     }
+    filePath = resolved.id
     url = pathToFileURL(resolved.id).toString()
   }
 
@@ -317,17 +318,17 @@ async function nodeImport(
 
   if (resolveOptions.legacyProxySsrExternalModules) {
     return proxyESM(mod)
-  } else if (isRuntimeHandled) {
-    return mod
-  } else {
+  } else if (filePath) {
     analyzeImportedModDifference(
       mod,
-      url,
+      filePath,
       id,
       metadata,
       resolveOptions.packageCache,
     )
     return proxyGuardOnlyEsm(mod, id)
+  } else {
+    return mod
   }
 }
 

